.github/workflows/code-style.yml

```
@@ -0,0 +1,73 @@
+  chk_coding_style:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 #v5.0.0
+
+      - name: Install dependencies
+        run: |
+          sudo apt -q update
+          sudo apt install -y shellcheck git
+
+      - name: Check for C++ coding style
+        id: style-check
+        run: |
+          output=$(scripts/check_style.sh 2>&1) || {
+            {
+              echo "output<<EOF"
+              echo "$output"
+              echo "EOF"
+            } >> $GITHUB_OUTPUT
+            exit 1
+          }
+
+      - name: Comment PR on failure
+        if: ${{ failure() && steps.style-check.outcome == 'failure' }}
+        uses: actions/github-script@ed597411d8f924073f98dfc5c65a23a2325f34cd #v8.0.0
+        with:
+          script: |
+            const output = `${{ steps.style-check.outputs.output }}`;
+            const body = `There was an error when running \`Code Style Check\` for commit \`${context.sha}\`:
+            \`\`\`
+            ${output}
+            \`\`\`
+            Please check that your changes are working as intended.`;
+
+            await github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: body
+            });
+
+            for (const line of output.split('\n')) {
+              const match = line.match(/^([^:]+\.(cpp|h)):(\d+):/);
+              if (!match) continue;
+              const [, filePath, , lineNumber] = match;
+              await github.rest.pulls.createReviewComment({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                pull_number: context.issue.number,
+                commit_id: context.sha,
+                path: filePath,
+                line: parseInt(lineNumber),
+                side: 'RIGHT',
+                body: 'Coding style error'
+              });
```

Maybe worth wrapping the createReviewComment calls in try/catch since they'll fail if the style error is in a file not part of the PR diff.

```yaml
  try {
    await github.rest.pulls.createReviewComment({...});
  } catch (e) {
    console.log(`Could not add inline comment for ${filePath}:${lineNumber}: ${e.message}`);
  }
```

---

Thanks for the suggestion! While I agree that wrapping `createReviewComment` in a try-catch is good defensive programming, I'm trying to understand the specific failure scenario.

The style checker script (`check_style.sh`) uses git grep to scan the entire repository for source files. However, since we run the style check on all PRs and direct pushes to `develop` are forbidden, the codebase should always be style-clean. This means any style errors found would be in files modified by the PR itself, and those lines would be part of the diff.

Could you clarify when this would fail? Maybe I'm missing something. But happy to add the try-catch regardless as a safety measure.